### DISCUSSION TITLE
Nested classes for MVU

### DISCUSSION THREAD

**Discussion by libin85** [dotnet/maui#83] [General]
I have been lately spending a lot of time exploring MVU architecture and Comet. Its pretty cool and fun to work with. One thing that really bothers me in almost all MVU style coding is how much code that goes into a single class file.

Can we have nested classes where in one we draw the UI, and in the other we write the logic. More like Xaml and code-behind. Infact also blazor has a better approach
Reference: https://www.telerik.com/blogs/using-a-code-behind-approach-to-blazor-components

**Comment by Happypig375** [comment]
>Can we have nested classes where in one we draw the UI, and in the other we write the logic.

It's called partial classes and you can already use them today.

**Comment by libin85** [comment]
> > Can we have nested classes where in one we draw the UI, and in the other we write the logic.
> 
> It's called partial classes and you can already use them today.

I totally get that, but I was asking if we could create like a nested class.  Like how xaml and the code behind creates together

**Comment by Happypig375** [comment]
>Like how xaml and the code behind creates together

It's also called "partial classes".

**Comment by libin85** [comment]
okay. did not know that.

**Comment by DingpingZhang** [comment]
If the "nested" you mentioned is the file structure shown below:

![image](https://user-images.githubusercontent.com/8541016/82668939-153e6380-9c6d-11ea-8915-28b4ca2aefea.png)

It is related to the configuration in the `*.csproj file`, not a C# syntax.

```xml
    <Compile Include="MainWindow.xaml.cs">
      <DependentUpon>MainWindow.xaml</DependentUpon>
      <SubType>Code</SubType>
    </Compile>
```

**Comment by charlesroddie** [comment]
You can also use a folder with two files.
